pyproject.toml

```diff
@@ -16,6 +16,7 @@ google-cloud-storage = "*"
 google-api-python-client = "*"
 google-auth = "*"
 google-auth-httplib2 = "*"
+boto3 = "^1.34.128"
```

Comment:
I'd prefer we leave this really open as "*" until there's a real conflict that requires us to limit it. Because this is a library imported  by other services, I want to be very flexible on the requirements until something actually requires limiting and we can't easily upgrade